packages/contracts-bedrock/test/L2/SuperchainWETH.t.sol

```diff
@@ -121,237 +130,189 @@ contract SuperchainWETH_Test is CommonTest {
+        uint256 _totalSupplyBefore = superchainWeth.totalSupply();
+        uint256 _toBalanceBefore = superchainWeth.balanceOf(_to);
+
+        // Look for the emit of the `Transfer` event
         vm.expectEmit(address(superchainWeth));
-        emit Transfer(_caller, address(0), _amount);
+        emit Transfer(ZERO_ADDRESS, _to, _amount);
+
+        // Look for the emit of the `CrosschainMinted` event
         vm.expectEmit(address(superchainWeth));
-        emit SendERC20(_caller, _recipient, _amount, _chainId);
-        vm.expectCall(Predeploys.ETH_LIQUIDITY, abi.encodeCall(IETHLiquidity.burn, ()), 1);
-        vm.expectCall(
-            Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER,
-            abi.encodeCall(
-                IL2ToL2CrossDomainMessenger.sendMessage,
-                (
-                    _chainId,
-                    address(superchainWeth),
-                    abi.encodeCall(superchainWeth.relayERC20, (_caller, _recipient, _amount))
-                )
-            ),
-            1
-        );
-        vm.prank(_caller);
-        superchainWeth.sendERC20(_recipient, _amount, _chainId);
+        emit CrosschainMinted(_to, _amount);
 
-        // Assert
-        assertEq(_caller.balance, 0);
-        assertEq(superchainWeth.balanceOf(_caller), 0);
+        // Mock the `isCustomGasToken` function to return false
+        _mockAndExpect(address(l1Block), abi.encodeCall(l1Block.isCustomGasToken, ()), abi.encode(false));
+
+        // Expect the call to the `mint` function in the `ETHLiquidity` contract
+        vm.expectCall(Predeploys.ETH_LIQUIDITY, abi.encodeCall(IETHLiquidity.mint, (_amount)), 1);
+
+        // Call the `mint` function with the bridge caller
+        vm.prank(Predeploys.SUPERCHAIN_TOKEN_BRIDGE);
+        superchainWeth.crosschainMint(_to, _amount);
+
+        // Check the total supply and balance of `_to` after the mint were updated correctly
+        assertEq(superchainWeth.totalSupply(), _totalSupplyBefore + _amount);
+        assertEq(superchainWeth.balanceOf(_to), _toBalanceBefore + _amount);
+        assertEq(address(superchainWeth).balance, _amount);
     }
 
-    /// @notice Tests that the sendERC20 function can be called with a sufficient balance on a
-    ///         custom gas token chain. Also tests that the proper calls are made and the proper
-    ///         events are emitted but ETH is not burned via the ETHLiquidity contract.
-    /// @param _amount The amount of WETH to send.
-    /// @param _chainId The chain ID to send the WETH to.
-    function testFuzz_sendERC20_sufficientFromCustomGasTokenChain_succeeds(uint256 _amount, uint256 _chainId) public {
-        // Assume
-        vm.assume(_chainId != block.chainid);
+    /// @notice Tests the `crosschainMint` with custom gas token succeeds and emits the `CrosschainMinted` event.
+    function testFuzz_crosschainMint_fromBridgeCustomGasTokenChain_succeeds(address _to, uint256 _amount) public {
```

Comment:
I think it'd be a good idea to use Foundry's `function expectCall(address callee, bytes calldata data, uint64 count) external;` vm cheatcode setting `count` to zero `mint` call over ETHLiquidity - so we ensure that's not called.

The same could be applied for `crosschainBurn()`